src/sdk/pynni/nni/nas/pytorch/cdarts/trainer.py

```diff
@@ -0,0 +1,203 @@
+import json
+import logging
+import os
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+import apex
+from apex.parallel import DistributedDataParallel
+from nni.nas.pytorch.cdarts import RegularizedDartsMutator, RegularizedMutatorParallel, DartsDiscreteMutator
+from nni.nas.pytorch.utils import AverageMeterGroup
+from utils import CyclicIterator, TorchTensorEncoder, accuracy, reduce_metrics
+
+PHASE_SMALL = "small"
+PHASE_LARGE = "large"
+
+class InteractiveKLLoss(nn.Module):
+    def __init__(self, temperature):
+        super().__init__()
+        self.temperature = temperature
+        # self.kl_loss = nn.KLDivLoss(reduction = 'batchmean')
+        self.kl_loss = nn.KLDivLoss()
+
+    def forward(self, student, teacher):
+        return self.kl_loss(F.log_softmax(student / self.temperature, dim=1),
+                            F.softmax(teacher / self.temperature, dim=1))
+
+
+class CdartsTrainer(object):
+    def __init__(self, model_small, model_large, criterion, loaders, samplers, logger, config):
```

Comment:
Please add docstring to trainer init method. Refer to [numpy doc](https://numpydoc.readthedocs.io/en/latest/format.html) for formats, and [here](https://github.com/microsoft/nni/blob/d0cf8e52e18d60a13e28c46d9333343280181dc2/src/sdk/pynni/nni/nas/pytorch/darts/trainer.py#L18) for an example.

Reply:
> Please add docstring to trainer init method. Refer to [numpy doc](https://numpydoc.readthedocs.io/en/latest/format.html) for formats, and [here](https://github.com/microsoft/nni/blob/d0cf8e52e18d60a13e28c46d9333343280181dc2/src/sdk/pynni/nni/nas/pytorch/darts/trainer.py#L18) for an example.

revised.